eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -1444,17 +1445,22 @@ registerRewardAccount = do
   khDelegator <- freshKeyHash
   registerStakeCredential (KeyHashObj khDelegator)
 
+tryLookupReward :: Credential 'Staking (EraCrypto era) -> ImpTestM era (Maybe Coin)
```

Comment:
I guess an alternative naming scheme would be `lookupReward` and `unsafeLookupReward`. But I based this naming on how we have named `submitTx` and `trySubmitTx`

Reply:
"unsafe" is a bad prefix in such scenario, because unsafe in Haskell usually describes things that violate some invariant and lead to dangerous situations like segfaults or other unpredictable behavior. Functions that throw errors when something goes wrong are called partial instead of unsafe. Considering that these function don't deal with anything unsafe there names should follow the common convention that is used in other libraries: 
* `lookup` to mean a total function that returns a `Maybe`
* `try` to mean a recovery from failure that returns an `Either`
* something like `get` or `partial` to indicate a runtime error. 

So, the names I propose are:
* change `lookup[Reward|KeyPair|...]` to indicate a total function that returns `Maybe`
* make partial functions prefixed with `get[Reward|KeyPair|..]`

I am ok with merging this PR with current naming and fixing all of this in a separate PR

Reply:
Created a ticket for this rename #4760